File: src/snippets/email-validation/reference/nodejs/DecisionLogAllow.js
import arcjet, { detectBot, validateEmail } from "@arcjet/node";
import express from "express";

const app = express();
const port = 3000;

app.use(express.urlencoded({ extended: false }));

const aj = arcjet({
  // Get your site key from https://app.arcjet.com and set it as an environment
  // variable rather than hard coding.
  key: process.env.ARCJET_KEY,
  rules: [
    validateEmail({
      mode: "LIVE",
      allow: ["DISPOSABLE"],
    }),
    detectBot({
      mode: "LIVE",
      allow: [], // "allow none" will block all detected bots
    }),
  ],
});

app.post("/", async (req, res) => {
  //const email = req.body.email;
  const email = "test@0zc7eznv3rsiswlohu.tk"; // Disposable email for demo
  console.log("Email received: ", email);

  const decision = await aj.protect(req, { email });
  console.log("Arcjet decision", decision);

  for (const result of decision.results) {
    console.log("Rule Result", result);

    if (result.reason.isEmail()) {
      console.log("Email rule", result);
    }

    if (result.reason.isBot()) {
      console.log("Bot protection rule", result);
    }
  }

  if (decision.isDenied()) {
    res.writeHead(403, { "Content-Type": "application/json" });
    res.end(JSON.stringify({ error: "Forbidden" }));
  } else {
    res.writeHead(200, { "Content-Type": "application/json" });
    res.end(JSON.stringify({ message: "Hello World", email }));
  }
});

app.listen(port, () => {
  console.log(`Example app listening on port ${port}`);
});

File: src/snippets/email-validation/reference/python-flask/DecisionLogAllow.py
import logging
import os

from arcjet import EmailType, Mode, arcjet_sync, detect_bot, validate_email
from flask import Flask, jsonify, request

app = Flask(__name__)

logger = logging.getLogger(__name__)

aj = arcjet_sync(
    key=os.environ["ARCJET_KEY"],  # Get your site key from https://app.arcjet.com
    rules=[
        validate_email(
            mode=Mode.LIVE,
            allow=[EmailType.DISPOSABLE],
        ),
        detect_bot(
            mode=Mode.LIVE,
            allow=[],  # "allow none" will block all detected bots
        ),
    ],
)


@app.post("/")
def index():
    # email = request.form.get("email", "")
    email = "test@0zc7eznv3rsiswlohu.tk"  # Disposable email for demo
    logger.info("Email received: %s", email)

    decision = aj.protect(request, email=email)
    logger.info("Arcjet decision %s", decision)

    for result in decision.results:
        logger.info("Rule Result %s", result)

        if result.reason_v2.type == "EMAIL":
            logger.info("Email rule %s", result)

        if result.reason_v2.type == "BOT":
            logger.info("Bot protection rule %s", result)

    if decision.is_denied():
        if decision.reason_v2.type == "EMAIL":
            return jsonify(error="Invalid email"), 400
        return jsonify(error="Forbidden"), 403

    return jsonify(message="Hello world", email=email)
